app/models/katello/glue/pulp/repo.rb

```diff
@@ -122,19 +123,15 @@ def create_pulp_repo
         raise PulpErrors::ServiceUnavailable.new(message, e)
       end
 
-      def generate_importer
+      def generate_importer(capsule = false)
         case self.content_type
         when Repository::YUM_TYPE
-          Runcible::Models::YumImporter.new(:ssl_ca_cert => self.feed_ca,
-                                            :ssl_client_cert => self.feed_cert,
-                                            :ssl_client_key => self.feed_key,
-                                            :feed => self.url,
-                                            :download_policy => self.download_policy)
+          Runcible::Models::YumImporter.new(yum_importer_values(capsule))
         when Repository::FILE_TYPE
           Runcible::Models::IsoImporter.new(:ssl_ca_cert => self.feed_ca,
                                             :ssl_client_cert => self.feed_cert,
                                             :ssl_client_key => self.feed_key,
-                                            :feed => self.url)
+                                            :feed => importer_feed_url(capsule))
         when Repository::PUPPET_TYPE
           options = {}
           options[:feed] = self.url if self.respond_to?(:url)
```

Comment:
I think you need to use importer_feed_url here too, otherwise i'm getting 'null' for the feed and the sync is failing.